### ISSUE TITLE
NativeAppEventEmitter.addListener('healthKit:Workout:new', cb) automatically triggers callback -- each time we open the app

### ISSUE THREAD

**Issue by aawolf1099** [agencyenterprise/react-native-health#236]
**Describe the bug**
**NativeAppEventEmitter.addListener('healthKit:Workout:new', cb) automatically triggers callback - Each time we open the app these listeners triggers which calls the http request all the time which is not ideal.
It should be called only once if there is any new data received.**

Anyone faced that too?

 

**Expected behavior**
Listeners should only be called if there is any new data

**Code**
 ```
useEffect(() => {
    async function handleAppleHealthNativeEvents() {
      let appleHealthTracking = await AsyncStorage.getItem(
        'appleHealthTracking',
      );
      if (JSON.parse(appleHealthTracking)) {
        NativeAppEventEmitter.addListener(
          'healthKit:Workout:setup:success',
          () => {},
        );
        NativeAppEventEmitter.addListener(
          'healthKit:Workout:setup:failure',
          () => console.log('Setup failure -->'),
        );
        NativeAppEventEmitter.addListener('healthKit:Workout:failure', () =>
          console.log('failure -->'),
        );
        NativeAppEventEmitter.addListener(
          'healthKit:Workout:new',
          appleHealthNewDataCallback,
        );
      }
    }
    if (Platform.OS === 'ios') {
      handleAppleHealthNativeEvents();
    }
  }, []);
```

 
**Smartphone (please complete the following information):**

- Device:  iPhone13 Pro Max

**Comment by devpascoe** [comment]
Getting the same. v1.10.0

**Comment by MartinThomasCW** [comment]
Same here on v.1.10.0.   Believe it doesn't work at all in later releases (awaiting fix in 1.14.0) - also the intent raise is causing my app to crash - but unsure why at present! (it runs the same code as if it was polling in the foreground and that never crashes at all!)

**Comment by GGGava** [comment]
Hey!

The listeners are built wrapping over HealthKit's [Observer Queries](https://developer.apple.com/documentation/healthkit/hkobserverquery/executing_observer_queries), which have this same behaviour, so this is actually caused by HealthKit itself, there isn't much we can do here.

Here you can find a similar discussion:
https://stackoverflow.com/questions/33139181/health-kit-observer-query-always-called-when-the-app-becomes-active 

I recommend checking if you have new data before doing anything else in the listener callback block.

I will close this issue but feel free to open a thread on the Discussion page if you want help or if you have a suggestion about this topic.
